includes/class-omise-callback.php

```diff
@@ -81,17 +93,6 @@ protected function invalid_result() {
 	 * Resolving a case of charge status: successful.
 	 */
 	protected function payment_successful() {
-		$message = __( 'OMISE: Payment successful.<br/>An amount of %1$s %2$s has been paid', 'omise' );
-
-		$this->order->payment_complete();
-		$this->order->add_order_note(
-			sprintf(
-				wp_kses( $message, array( 'br' => array() ) ),
-				$this->order->get_total(),
-				$this->order->get_order_currency()
-			)
-		);
-
```

Comment:
Note: moving all the code that is used for manipulating the order status into `Omise_Payment_Result` class. Leaving only those UI-related code here in the `Omise_Callback` class. 

As in a concept of, `Omise_Callback` would be only a route for buyer to be redirected back, then get resolved to a proper page based on the result of a particular charge.